### PR TITLE
PCHR-791: Fixed algorithm for calculating the length of service

### PR DESCRIPTION
**Pull request by Agares4**
The length of service on "Contact summary" page wasn't calculated properly, so in case of no contracts that were both primary and current, invalid value was displayed. 
My fix includes:
1. properly handling case of absence of primary contract
2. Improving algorithm for finding primary contract

### CODE REVIEW

**Comment by AkA84** [comment]
Missing ticket number in both the PR title and most importantly on the commit subject line

File: contactsummary/js/src/contact-summary/services/contract.js [hunk 1]
```diff
@@ -75,15 +75,9 @@ define([
          */
         factory.getPrimary = function () {
             return this.get().then(function (response) {
-                var primary = {};
+                var sortedContracts = _.sortBy(response, ['end_date', 'is_primary'], ['desc', 'desc']);
 
-                angular.forEach(response, function (contract) {
-                    if (contract.is_primary === '1' && contract.is_current === '1') {
```
Comment (AkA84) [inline contactsummary/js/src/contact-summary/services/contract.js]: What about the check on `is_current`?

File: contactsummary/js/src/contact-summary/controllers/keyDetails.js [hunk 2]
```diff
@@ -26,6 +26,10 @@ define([
                 return Contract.getPrimary();
             }.bind(this))
             .then(function (response) {
+                if (_.isEmpty(response)) {
+                    return;
```
Comment (AkA84) [inline contactsummary/js/src/contact-summary/controllers/keyDetails.js]: So what happens if there is no primary contract?

File: contactsummary/js/src/contact-summary/services/contract.js (hunk 1 above)
Comment (Agares4) [inline contactsummary/js/src/contact-summary/services/contract.js]: according to comments on JIRA task, the most recent contract should be used here, even if it's not current

File: contactsummary/js/src/contact-summary/controllers/keyDetails.js (hunk 2 above)
Comment (Agares4) [inline contactsummary/js/src/contact-summary/controllers/keyDetails.js]: Fields in view are not populated.

File: contactsummary/js/src/contact-summary/services/contract.js (hunk 1 above)
Comment (AkA84) [inline contactsummary/js/src/contact-summary/services/contract.js]: Ok, please add document what constitutes a primary contract in the function description

File: contactsummary/js/src/contact-summary/controllers/keyDetails.js (hunk 2 above)
Comment (AkA84) [inline contactsummary/js/src/contact-summary/controllers/keyDetails.js]: Yeah, but there is no fallback? No default message or anything?

Comment (Agares4) [inline contactsummary/js/src/contact-summary/controllers/keyDetails.js]: I followed the convention (all of the fields in contact summary have empty defaults). I can add some default if needed.

**Comment by AkA84** [comment]
Approved

**Pull request merged**